### PR TITLE
Fix Shared Database Tests

### LINKED ISSUE #6862: Fix shared database tests
Currently, in the latest master version the SharedDatabaseTests are failing with the following error

```
org.jabref.logic.shared.SynchronizationTestSimulator

  Test simulateUpdateOnNoLongerExistingEntry() FAILED

  org.opentest4j.AssertionFailedError: expected: not <null>
      at org.jabref.logic.shared.SynchronizationTestSimulator.simulateUpdateOnNoLongerExistingEntry(SynchronizationTestSimulator.java:140)

  Test simulateEntryChangeConflicts() FAILED

  org.opentest4j.AssertionFailedError: expected: not <null>
      at org.jabref.logic.shared.SynchronizationTestSimulator.simulateEntryChangeConflicts(SynchronizationTestSimulator.java:165)

```

To execute the tests locally you need to setup a shared SQL Databse, PostgresSQL 
The postgresSQL must run on localhost with the following setup:
DBname: postgres
User: postgres
Pasword: postgres

Alternatively change the values in the class `TestConnector`, method  `DBMSConnectionProperties getTestConnectionProperties()`

### PR DESCRIPTION
**Pull request by Siedlerchr**
Fix FieldChangeEvent Delta computation

The CoarseChangeFilter only reports delta >0 as Major change.
The problem is that when you codewise set new field content with the same exact length as before, but not the same content it still has a delta of 0 and therefore no DB sync is triggered. 
A more detailed explanation is https://github.com/JabRef/jabref/issues/6862#issuecomment-716028639

Fixes #6862 

- [ ] Change in CHANGELOG.md described (if applicable)
- [x] Tests created for changes (if applicable)
- [x] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for UI changes)
- [ ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, submitted a pull request to the documentation repository.

### CODE REVIEW

File: src/main/java/org/jabref/model/entry/event/FieldChangedEvent.java [hunk 1]
```diff
@@ -61,10 +61,12 @@ public FieldChangedEvent(FieldChange fieldChange) {
     private int computeDelta(String oldValue, String newValue) {
         if (oldValue == newValue) {
             return 0;
-        } else if (oldValue == null && newValue != null) {
+        } else if ((oldValue == null) && (newValue != null)) {
             return newValue.length();
-        } else if (newValue == null && oldValue != null) {
+        } else if ((newValue == null) && (oldValue != null)) {
             return oldValue.length();
+        } else if ((Math.abs(newValue.length() - oldValue.length()) == 0) && !oldValue.equals(newValue)) {
```
Comment (tobiasdiez) [inline src/main/java/org/jabref/model/entry/event/FieldChangedEvent.java:68]: why not `newValue.length() == oldValue.length()` ?

File: src/main/java/org/jabref/model/entry/event/FieldChangedEvent.java [hunk 2]
```diff
@@ -61,10 +61,12 @@ public FieldChangedEvent(FieldChange fieldChange) {
...
+            return newValue.length();
```
Comment (tobiasdiez) [inline src/main/java/org/jabref/model/entry/event/FieldChangedEvent.java:69]: How is the `delta` value used? Maybe we need to calculate the actual difference between the strings here? I.e. the delta of `aaaa` and `aabb` is 2 and not 4

**Review by tobiasdiez — CHANGES_REQUESTED** [review]

File: src/main/java/org/jabref/model/entry/event/FieldChangedEvent.java (hunk 1 above)
Comment (Siedlerchr) [reply src/main/java/org/jabref/model/entry/event/FieldChangedEvent.java:68]: Yeah, that makes sense

File: src/main/java/org/jabref/model/entry/event/FieldChangedEvent.java (hunk 2 above)
Comment (Siedlerchr) [reply src/main/java/org/jabref/model/entry/event/FieldChangedEvent.java:69]: @tobiasdiez  The delta value is only used here to check if it's >1
https://github.com/JabRef/jabref/blob/a0ca87530991900db5b34df3683aa010dffe97c8/src/main/java/org/jabref/logic/util/CoarseChangeFilter.java#L49-L51

Comment (tobiasdiez) [reply src/main/java/org/jabref/model/entry/event/FieldChangedEvent.java:69]: Ok, thanks for looking this up. Then I would propose to replace `delta` by `isMajorChange`.

**Pull request merged**